packages/nuxt/src/runtime/plugins/database.server.ts

```diff
@@ -0,0 +1,205 @@
+import {
+  type Span,
+  type StartSpanOptions,
+  addBreadcrumb,
+  captureException,
+  debug,
+  flushIfServerless,
+  SEMANTIC_ATTRIBUTE_SENTRY_OP,
+  SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN,
+  SPAN_STATUS_ERROR,
+  startSpan,
+} from '@sentry/core';
+import type { Database, PreparedStatement } from 'db0';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { defineNitroPlugin, useDatabase } from 'nitropack/runtime';
+// @ts-expect-error - This is a virtual module
+import { databaseInstances } from '#sentry/database-config.mjs';
+
+/**
+ * Keeps track of prepared statements that have been patched.
+ */
+const patchedStatement = new WeakSet<PreparedStatement>();
+
+/**
+ * The Sentry origin for the database plugin.
+ */
+const SENTRY_ORIGIN = 'auto.db.nuxt';
+
+/**
+ * Creates a Nitro plugin that instruments the database calls.
+ */
+export default defineNitroPlugin(() => {
+  try {
+    debug.log('@sentry/nuxt: Instrumenting databases...');
+
+    for (const instance of databaseInstances) {
+      debug.log('@sentry/nuxt: Instrumenting database instance:', instance);
+      const db = useDatabase(instance);
+      instrumentDatabase(db);
```

Comment:
I think this only runs once. But if it doesn't: Maybe we need a conditional here to check whether the database was already instrumented.